### ISSUE TITLE
Some modules (e.g. argparse) accept ascii-encodable unicode values in Python 2, but their types don't reflect this

### ISSUE THREAD

**Issue by posita** [python/typeshed#756]
Consider:

```py
# validates_and_works_in_py2_py3.py
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('str')  # a unicode literal in Python 3, bytes in Python 2
parser.parse_args()
```

```
$ python2.7 validates_and_works_in_py2_py3.py foo
$ python3.4 validates_and_works_in_py2_py3.py foo
$ mypy validates_and_works_in_py2_py3.py
$ mypy --py2 validates_and_works_in_py2_py3.py
```

```py
# validates_only_in_py3_but_works_in_py2_py3.py
from __future__ import unicode_literals
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('str')  # now a unicode literal in both Python 2 and 3
parser.parse_args()
```

```
$ python2.7 validates_only_in_py3_but_works_in_py2_py3.py foo
$ python3.4 validates_only_in_py3_but_works_in_py2_py3.py foo
$ mypy validates_only_in_py3_but_works_in_py2_py3.py
$ mypy --py2 validates_only_in_py3_but_works_in_py2_py3.py
validates_only_in_py3_but_works_in_py2_py3.py:4: error: Argument 1 to "add_argument" of "ArgumentParser" has incompatible type "unicode"; expected Sequence[str]
```

Consider changing `argparse.pyi` (and similarly-affected modules) to support `unicode` or something like `Union[str, unicode]`. (Thanks @JukkaL, for the guidance!)

**Comment by posita** [comment]
FYI, it looks like `logging.pyi` might be similar affected. Consider, e.g., `logging.getLogger`.

**Comment by JukkaL** [comment]
Fixed in #842.

**Comment by JukkaL** [comment]
(In particular, I fixed the issue in argparse. I'll create a separate issue for logging.)
